### DISCUSSION TITLE
IPv6 connections refused

### DISCUSSION THREAD

**Discussion by Aldaviva** [testspace-com/support#30] [Q&A] [answered]
Hello,

HTTPS connections to `*.testspace.com` are being refused over IPv6, but they succeed with the IPv4 address. This seemed to start happening in July 2023. It's happening for other users as well, not just those using the same user agent, LAN, ISP, or DNS servers as me.

Some user agents like browsers and curl can automatically fall back to IPv4, but not all clients have this behavior, leading to unrecoverable errors. A client-side workaround would be to only use the A record, but a server-side fix to either accept IPv6 requests or remove the AAAA record would benefit all clients (with improved performance, at the very least).

Thanks for looking at this!

### IPv6 → IPv4 fallback in curl
```text
$ curl -v --head https://aldaviva.testspace.com
* processing: https://aldaviva.testspace.com
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying [2600:1f14:569:e000:d60:2c6a:8d0c:9502]:443...
*   Trying 54.202.124.205:443...
* Connected to aldaviva.testspace.com (54.202.124.205) port 443
* schannel: disabled automatic use of client certificate
* using HTTP/1.x
> HEAD / HTTP/1.1
> Host: aldaviva.testspace.com
> User-Agent: curl/8.2.1
> Accept: */*
>
< HTTP/1.1 200 OK
```

### IPv6 only connection refused in curl
```text
$ curl -6 -v https://aldaviva.testspace.com
* processing: https://aldaviva.testspace.com
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying [2600:1f14:569:e000:d60:2c6a:8d0c:9502]:443...
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0* connect to 2600:1f14:569:e000:d60:2c6a:8d0c:9502 port 443 failed: Connection refused
* Failed to connect to aldaviva.testspace.com port 443 after 2174 ms: Couldn't connect to server
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
* Closing connection
curl: (7) Failed to connect to aldaviva.testspace.com port 443 after 2174 ms: Couldn't connect to server
```

**Comment by ivailop** [comment]
Could you please try with any other (non Testspace) site? 

Using my home wi-fi, had the same problem with any site (e.g. google.com)! Switching to cell-phone-hotspot it worked all fine. 
My home internet provider is Spectrum...

**Reply by Aldaviva** [reply]
Sure, I am able to access other IPv6 sites fine. Here are some test results:

### [test-ipv6.com](https://test-ipv6.com)
![image](https://github.com/testspace-com/support/assets/1417794/67b82d94-dd79-457b-a59b-57324bf2027a)

### [Google IPv6 test](https://ipv6test.google.com)
![image](https://github.com/testspace-com/support/assets/1417794/6793b650-0147-4224-a073-93cdcb8c5d1e)

### [Google](https://google.com)
![image](https://github.com/testspace-com/support/assets/1417794/17df10bf-d58b-41cc-b98b-5b7ed6d86520)

In addition, this issue is not isolated to me or my ISP, as I mentioned:
> It's happening for other users as well, not just those using the same user agent, LAN, ISP, or DNS servers as me.

**Reply by ivailop** [reply, marked as answer]
We identified the core problem and we believe we addressed it. Please try again.

**Reply by Aldaviva** [reply]
Thanks, it's working for me now.

I appreciate your help!

```text
$ curl -6 -v --head https://aldaviva.testspace.com
* processing: https://aldaviva.testspace.com
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying [2600:1f14:569:e000:d60:2c6a:8d0c:9502]:443...
* Connected to aldaviva.testspace.com (2600:1f14:569:e000:d60:2c6a:8d0c:9502) port 443
* schannel: disabled automatic use of client certificate
* using HTTP/1.x
> HEAD / HTTP/1.1
> Host: aldaviva.testspace.com
> User-Agent: curl/8.2.1
> Accept: */*
>
< HTTP/1.1 200 OK
```
